ast/dml.go

```diff
@@ -1519,7 +1519,41 @@ type FrameBound struct {
 
 // Restore implements Node interface.
 func (n *FrameBound) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	if n.UnBounded {
+		ctx.WriteKeyWord("UNBOUNDED")
+	}
+	switch n.Type {
+	case CurrentRow:
+		ctx.WriteKeyWord("CURRENT ROW")
+	case Preceding, Following:
+		if n.Unit != nil {
+			ctx.WriteKeyWord("INTERVAL ")
+		}
+		if n.Expr != nil {
+			if err := n.Expr.Restore(ctx); err != nil {
+				return errors.Annotate(err, "An error occurred while restore FrameBound.Expr")
+			}
+		}
+		if n.Unit != nil {
+			// Here the Unit string should not be quoted
+			singleQuotesFlag := ctx.Flags & RestoreStringSingleQuotes
```

Comment:
My temporary plan:
```
var sb string.Builder
n.Unit.Restore(NewRestoreCtx(DefaultRestoreFlags, &sb))
ctx.WriteKeyWord(sb.String())
```

Reply:
Both is fine as a workaround, and we need to add a TODO comment improving this no matter which we choose 🙃

Reply:
TODO what? make a unit expression and change the TiDB?

Reply:
@leoppro Yes.